webhooks/terminate.js

```diff
@@ -0,0 +1,65 @@
+// @ts-check
+const router = require('express').Router();
+const jose = require('jose');
+const crypto = require('crypto');
+
+const config = require('../lib/config');
+const logger = require('../lib/logger');
+
+/**
+ * This is a webhook that can be called when infrastructure wants this instance
+ * of the app to terminate. This is useful e.g. as part of deploy tooling. It
+ * will send SIGTERM to the current process, which will cause it to exit
+ * gracefully.
+ *
+ * This app requires a signed `PrairieLearn-Signature` JWT header to be present,
+ * so it's safe to publicly expose this endpoint if needed. However, it's
+ * recommended that this endpoint be blocked at your load balancer or firewall
+ * for extra security.
+ */
+router.post('/', async (req, res) => {
+  try {
+    const jwt = req.headers['prairielearn-signature'];
```

Comment:
I know we usually use our CSRF token machinery when we want some kind of signed data, but I'm opting to use a standard JWT here. It's easier to generate/verify them from outside PL without needing to copy/paste around our custom code. Long term, I think we should broadly replace our custom CSRF tokens with JWTs.

Reply:
Yep, I'm on board with this. I assume that the standard JWT libraries no longer have the [trivial vulnerabilities](https://auth0.com/blog/critical-vulnerabilities-in-json-web-token-libraries/) that were rife in the early versions.

Reply:
Yeah, I think the fact we are using symmetric key signing makes all these attacks irrelevant.